### PR TITLE
Use a venv for python

### PR DESCRIPTION
**Pull request by 2uasimojo**
...so we can install modules as non-root.

### CODE REVIEW

**Comment by 2uasimojo** [comment]
We're going to move the venv construction into ensure.sh

/hold

**Comment by 2uasimojo** [comment]
Tested. Works.

/hold cancel

File: boilerplate/_lib/ensure.sh [hunk 1]
```diff
@@ -22,6 +22,14 @@ golangci-lint)
         curl -sfL "${DOWNLOAD_URL}" | tar -C "${GOPATH}/bin" -zx --strip-components=1 "golangci-lint-${GOLANGCI_LINT_VERSION}-${GOOS}-amd64/golangci-lint"
     fi
     ;;
+venv)
+    # Set up a python virtual environment
+    python3 -m venv .venv
```
Comment (jharrington22) [inline boilerplate/_lib/ensure.sh:27]: Where is this being created? I thought you didn't have permissions in the directory the user was running?

Comment (2uasimojo) [reply boilerplate/_lib/ensure.sh:27]: The pwd is the repo root

Comment (jharrington22) [reply boilerplate/_lib/ensure.sh:27]: ah :+1:

Comment (2uasimojo) [reply boilerplate/_lib/ensure.sh:27]: ...based on having run this via a `make` from that directory.

Comment (dustman9000) [inline boilerplate/_lib/ensure.sh:27]: We should be ensuring the python3 is installed explicitly and obtain the absolute path to it before attempting the venv install.

Comment (2uasimojo) [reply boilerplate/_lib/ensure.sh:27]: Agree. That should be covered (along with installing the libs) as part of https://issues.redhat.com/browse/OSD-5074 / https://issues.redhat.com/browse/OSD-5096 / etc.

**Pull request merged**